baseplate/lib/edge_context.py

```diff
@@ -215,17 +215,37 @@ def has_role(self, role: str) -> bool:
 
     def event_fields(self) -> Dict[str, Any]:
         """Return fields to be added to events."""
-        if self.is_logged_in:
-            user_id = self.id
-        else:
-            user_id = self.loid
+        loid: Optional[str] = self.loid
+        if loid == "":
+            loid = None
 
         return {
-            "user_id": user_id,
+            "user_id": loid,
             "logged_in": self.is_logged_in,
             "cookie_created_timestamp": self.cookie_created_ms,
         }
 
+    @property
+    def loid(self) -> str:
+        # First, if it's logged in user, return logged in user id.
+        try:
+            user_id = self.id
+            if user_id is not None:
+                return user_id
+        except NoAuthenticationError:
+            pass
+
+        # Next, return the loid from thrift payload if it's non-empty
+        if self.loid_ != "":
```

Comment:
maybe

```suggestion
        if self.loid_:
```
?

technically None != ""